### DISCUSSION TITLE
How to use machines created with model of createModel and @xstate/react

### DISCUSSION THREAD

**Discussion by bylethviva** [statelyai/xstate#2481] [Q&A] [answered]
I've been exploring `createModel` and so far I really enjoy the convenience that it creates for typings and sending actions!

Is it possible to use the machine created with `model.createMachine` with `useMachine` and still have the convenience of sending events via the model? Should I be using a different hook instead to get this to work?

**Comment by davidkpiano** [comment, marked as answer]
Glad you enjoy it!

It is possible; if you export the model, you can import it and use its event creators:

```js
import { someModel } from '../someModel';

// ...

// In a component:
send(someModel.events.greet('Hello'));

```

**Reply by bylethviva** [reply]
I like how that flows a lot! I think it helps to illustrate where the event is coming from due to the use of the model. 

I've got a second more bike shed related question. Previously I was using all caps for my event type names. Do you think that's necessary anymore given the fact that you end up accessing the event from the events property?

**Reply by ChrisShank** [reply]
Ya you don't have to name events in all caps if you don't want to! Personally I have been enjoying using spaces for states/events names as its really readable in the visualizer!

```ts
createMachine({
  initial: 'some state',
  states: {
    'some state': {
      on: {
        'some event': 'some other state',
      }
    },
    'some other state': {},
  }
})
```
